src/pytest_plugins/filler/gen_test_doc/gen_test_doc.py

```diff
@@ -0,0 +1,543 @@
+    if config.getoption("--gen-docs"):
+        config.option.disable_html = True
+        config.pluginmanager.register(TestDocsGenerator(config), "test-case-doc-generator")
+
+
+def get_test_function_id(item: Item) -> str:
+    """
+    Get the test function's ID from the item.
+    """
+    return item.nodeid.split("[")[0]
+
+
+def get_test_function_name(item: Item) -> str:
+    """
+    Get the test function's name from the item.
+    """
+    return item.name.split("[")[0]
+
+
+def get_test_case_id(item: Item) -> str:
+    """
+    Get the test case's ID from the item.
+    """
+    return item.nodeid.split("[")[-1].rstrip("]")
+
+
+def get_test_function_import_path(item: pytest.Item) -> str:
+    """
+    Retrieve the fully qualified import path for an item's test function.
+
+    This is used in jinja2 templates to get the test function, respectively
+    the test function's class, documentation with mkdocstrings.
+    """
+    item = cast(pytest.Function, item)  # help mypy infer type
+    module_name = item.module.__name__
+    if hasattr(item.obj, "__self__"):
+        # it's a method bound to a class
+        test_class = item.obj.__self__.__class__.__name__
+        test_function = item.obj.__name__
+        full_path = f"{module_name}.{test_class}"
+    else:
+        # it's a standalone function, no class
+        test_function = item.obj.__name__
+        full_path = f"{module_name}.{test_function}"
+    return full_path
+
+
+def get_import_path(path: Path) -> str:
+    """
+    Get the import path for a given path.
+
+    - For modules, strip the file extension.
+    - For directories (i.e., packages such as `tests.berlin`), `with_suffix()` is ignored.
+
+    To do:
+    ------
+
+    - This should be combined with `get_test_function_import_path`.
+    """
+    return str(path.with_suffix("")).replace("/", ".")
+
+
+def create_github_issue_url(title: str) -> str:
+    """
+    Create a GitHub issue URL for the given title.
+    """
+    url_base = "https://github.com/ethereum/execution-spec-tests/issues/new?"
+    title = title.replace(" ", "%20")
+    labels = "scope:docs,type:bug"
+    return f"{url_base}title={title}&labels={labels}"
+
+
+def get_docstring_one_liner(item: pytest.Item) -> str:
+    """
+    Extracts either the first 100 characters or the first line of the docstring
+    from the function associated with the given pytest.Item.
+    """
+    item = cast(pytest.Function, item)  # help mypy infer type
+    func_obj = item.obj
+    docstring = func_obj.__doc__
+    test_function_name = get_test_function_name(item)
+
+    if not docstring:
+        github_issue_url = create_github_issue_url(
+            f"docs(bug): No docstring available for `{test_function_name}`"
+        )
+        logger.warning(f"No docstring available for `{test_function_name}`.")
+        return f"[📖🐛 No docstring available]({github_issue_url})"
+    docstring = docstring.strip()
+    lines = docstring.splitlines()
```

Comment:
I realized on 7702 tests that I copy-pasted the docstrings of some of the tests.

I think this would be a good time to check for duplicate docstrings in different test functions:

```suggestion
    test_function_id = get_test_function_id(item)
    if (
        docstring in docstring_test_function_history
        and docstring_test_function_history[docstring] != test_function_id
    ):
        logger.warning(
            f"Duplicate docstring for {test_function_id}: "
            f"{docstring_test_function_history[docstring]} and {test_function_id}"
        )
    else:
        docstring_test_function_history[docstring] = test_function_id
    lines = docstring.splitlines()
```

`docstring_test_function_history` in this case is a simple global dictionary to keep track of the docstring texts of all test functions visited so far.

Reply:
Great additional check.

Reply:
Oof, more duplicates detected than expected. Left the check in this PR but reduced the log level from `warning` to `info` for now.

Will add in a separate PR. Issue: #828.